taxcalc/utils.py

```diff
@@ -192,9 +192,12 @@ def add_weighted_income_bins(pdf, num_bins=10, labels=None,
                                                    pdf['s006'].values))
     else:
         pdf['cumsum_temp'] = np.cumsum(pdf['s006'].values)
+    min_cumsum = pdf['cumsum_temp'].values[0]
     max_cumsum = pdf['cumsum_temp'].values[-1]
-    bin_edges = [0] + list(np.arange(1, (num_bins + 1)) *
-                           (max_cumsum / float(num_bins)))
+    cumsum_range = max_cumsum - min_cumsum
+    bin_width = cumsum_range / float(num_bins)
+    bin_edges = [-9e99] + list(np.arange(1, (num_bins + 1)) * bin_width)
+    bin_edges[-1] = 9e99  # raise top of last bin to include all observations
```

Comment:
Looking at the 4 lines here, I'm not sure whether it will do the job or not. I have not tested it on my local but will do in a few minutes. Use random number for logic walk-through -- suppose max_cumsum is 165 million when it's not weighted by income_measure, then bin_width would be 16.5 million if we have 10 bins. Then all the bin_widths are stacked on top of negative infinity [-9e99] for bin edges. I don't think 165 million would make any bin edge positive before changing the last one to positive infinity [9e99].